### PR TITLE
feat: add $notnil hook

### PR DESCRIPTION
**Pull request by jhesgodi**
image

### CODE REVIEW

File: src/classnames.ts [hunk 1]
```diff
@@ -61,19 +63,29 @@ export function get(
...
 
   // at this stage we're dealing with prop -> a range of variations (this is an object)
   // we need to resolve the variation that match current value of the prop
   // if value of the prop doesn't have a match, we should divert to $none hook here.
-  const matchingVariations = valueVariations[correspondingSourceValue.toString()] || valueVariations[$none];
+  let matchingVariations = valueVariations[correspondingSourceValue.toString()] || valueVariations[$none];
+
+  // when $notnil was provided and prop value is not nil we append $nonil value to matching variations
+  // have to filter undefined in case there were no matches
+  if (isNil(valueVariations[$notnil]) === false && isNil(correspondingSourceValue) === false) {
```
Comment (mattvalley4iconic) [inline src/classnames.ts]: `isNil(valueVariations[$notnil]) === false` -> `!isNil(valueVariations[$notnil])`

Comment (mattvalley4iconic) [reply src/classnames.ts]: `isNil(correspondingSourceValue) === false` -> `!isNil(correspondingSourceValue)`

**Review by mattvalley4iconic — CHANGES_REQUESTED** [review]

File: src/classnames.ts [hunk 2]
```diff
@@ -93,7 +105,9 @@ export function get(
     }
 
     // prop value (or $none) maps to string, if we don't have $always just return what we got
-    if (isNil(alwaysVar)) return matchingVariations;
+    if (isNil(alwaysVar) === true) {
```
Comment (mattvalley4iconic) [inline src/classnames.ts]: do we need this change? 🤔

Comment (jhesgodi) [reply src/classnames.ts]: I was updating the if blocks to be blocks
`if (something) return` :x:
`if (something) { return }` ✔️

**Review by mattvalley4iconic — APPROVED** [review]

**Pull request merged**